File: sdks/thunkmetrc-java/wrapper/src/main/java/io/github/thunkier/thunkmetrc/wrapper/models/Hub.java
package io.github.thunkier.thunkmetrc.wrapper.models;import java.util.List;
import com.fasterxml.jackson.annotation.JsonProperty;public class Hub {
    @JsonProperty("ActualArrivalDateTime")
    public String actualArrivalDateTime;
    @JsonProperty("ActualDepartureDateTime")
    public String actualDepartureDateTime;
    @JsonProperty("ActualReturnArrivalDateTime")
    public String actualReturnArrivalDateTime;
    @JsonProperty("ActualReturnDepartureDateTime")
    public String actualReturnDepartureDateTime;
    @JsonProperty("CreatedByUserName")
    public String createdByUserName;
    @JsonProperty("CreatedDateTime")
    public String createdDateTime;
    @JsonProperty("DeliveryCount")
    public Integer deliveryCount;
    @JsonProperty("DeliveryId")
    public Integer deliveryId;
    @JsonProperty("DeliveryPackageCount")
    public Integer deliveryPackageCount;
    @JsonProperty("DeliveryReceivedPackageCount")
    public Integer deliveryReceivedPackageCount;
    @JsonProperty("DriverName")
    public String driverName;
    @JsonProperty("DriverOccupationalLicenseNumber")
    public String driverOccupationalLicenseNumber;
    @JsonProperty("DriverVehicleLicenseNumber")
    public String driverVehicleLicenseNumber;
    @JsonProperty("EstimatedArrivalDateTime")
    public String estimatedArrivalDateTime;
    @JsonProperty("EstimatedDepartureDateTime")
    public String estimatedDepartureDateTime;
    @JsonProperty("EstimatedReturnArrivalDateTime")
    public String estimatedReturnArrivalDateTime;
    @JsonProperty("EstimatedReturnDepartureDateTime")
    public String estimatedReturnDepartureDateTime;
    @JsonProperty("Id")
    public Integer id;
    @JsonProperty("IsLayover")
    public Boolean isLayover;
    @JsonProperty("IsVoided")
    public Boolean isVoided;
    @JsonProperty("LastModified")
    public String lastModified;
    @JsonProperty("ManifestNumber")
    public String manifestNumber;
    @JsonProperty("PackageCount")
    public Integer packageCount;
    @JsonProperty("ReceivedDateTime")
    public String receivedDateTime;
    @JsonProperty("ReceivedDeliveryCount")
    public Integer receivedDeliveryCount;
    @JsonProperty("ReceivedPackageCount")
    public Integer receivedPackageCount;
    @JsonProperty("RecipientFacilityLicenseNumber")
    public String recipientFacilityLicenseNumber;
    @JsonProperty("RecipientFacilityName")
    public String recipientFacilityName;
    @JsonProperty("RejectedPackagesReturned")
    public Boolean rejectedPackagesReturned;
    @JsonProperty("ShipmentTransactionType")
    public Integer shipmentTransactionType;
    @JsonProperty("ShipmentTransporterDetails")
    public List<TransfersHubShipmentTransporterDetailsItem> shipmentTransporterDetails;
    @JsonProperty("ShipmentTypeName")
    public String shipmentTypeName;
    @JsonProperty("ShipperFacilityLicenseNumber")
    public String shipperFacilityLicenseNumber;
    @JsonProperty("ShipperFacilityName")
    public String shipperFacilityName;
    @JsonProperty("TransporterAcceptedDateTime")
    public String transporterAcceptedDateTime;
    @JsonProperty("TransporterActualArrivalDateTime")
    public String transporterActualArrivalDateTime;
    @JsonProperty("TransporterActualDepartureDateTime")
    public String transporterActualDepartureDateTime;
    @JsonProperty("TransporterEstimatedArrivalDateTime")
    public String transporterEstimatedArrivalDateTime;
    @JsonProperty("TransporterEstimatedDepartureDateTime")
    public String transporterEstimatedDepartureDateTime;
    @JsonProperty("TransporterFacilityLicenseNumber")
    public String transporterFacilityLicenseNumber;
    @JsonProperty("TransporterFacilityName")
    public String transporterFacilityName;
    @JsonProperty("VehicleLicensePlateNumber")
    public String vehicleLicensePlateNumber;
    @JsonProperty("VehicleMake")
    public String vehicleMake;
    @JsonProperty("VehicleModel")
    public String vehicleModel;
    public static class TransfersHubShipmentTransporterDetailsItem {
    @JsonProperty("ActualDriverStartDateTime")
    public String actualDriverStartDateTime;
    @JsonProperty("DriverLayoverLeg")
    public String driverLayoverLeg;
    @JsonProperty("DriverName")
    public String driverName;
    @JsonProperty("DriverOccupationalLicenseNumber")
    public String driverOccupationalLicenseNumber;
    @JsonProperty("DriverVehicleLicenseNumber")
    public String driverVehicleLicenseNumber;
    @JsonProperty("ShipmentDeliveryId")
    public Integer shipmentDeliveryId;
    @JsonProperty("VehicleLicensePlateNumber")
    public String vehicleLicensePlateNumber;
    @JsonProperty("VehicleMake")
    public String vehicleMake;
    @JsonProperty("VehicleModel")
    public String vehicleModel;
}
}

File: sdks/thunkmetrc-py/wrapper/src/thunkmetrc/wrapper/models/hub.py
from typing import TypedDict, List, Optional, Any, Generic, TypeVar, TYPE_CHECKING
from typing_extensions import TypedDict

if TYPE_CHECKING:
    from . import *

class Hub(TypedDict, total=False):
    ActualArrivalDateTime: str
    ActualDepartureDateTime: str
    ActualReturnArrivalDateTime: str
    ActualReturnDepartureDateTime: str
    CreatedByUserName: str
    CreatedDateTime: str
    DeliveryCount: int
    DeliveryId: int
    DeliveryPackageCount: int
    DeliveryReceivedPackageCount: int
    DriverName: str
    DriverOccupationalLicenseNumber: str
    DriverVehicleLicenseNumber: str
    EstimatedArrivalDateTime: str
    EstimatedDepartureDateTime: str
    EstimatedReturnArrivalDateTime: str
    EstimatedReturnDepartureDateTime: str
    Id: int
    IsLayover: bool
    IsVoided: bool
    LastModified: str
    ManifestNumber: str
    PackageCount: int
    ReceivedDateTime: str
    ReceivedDeliveryCount: int
    ReceivedPackageCount: int
    RecipientFacilityLicenseNumber: str
    RecipientFacilityName: str
    RejectedPackagesReturned: bool
    ShipmentTransactionType: int
    ShipmentTransporterDetails: List['TransfersHubShipmentTransporterDetailsItem']
    ShipmentTypeName: str
    ShipperFacilityLicenseNumber: str
    ShipperFacilityName: str
    TransporterAcceptedDateTime: str
    TransporterActualArrivalDateTime: str
    TransporterActualDepartureDateTime: str
    TransporterEstimatedArrivalDateTime: str
    TransporterEstimatedDepartureDateTime: str
    TransporterFacilityLicenseNumber: str
    TransporterFacilityName: str
    VehicleLicensePlateNumber: str
    VehicleMake: str
    VehicleModel: str
